app/server_inactivity.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+	"github.com/mattermost/mattermost-server/v6/store"
+)
+
+const serverInactivityHours = 100
+
+func (s *Server) doInactivityCheck() {
+	if !s.Config().FeatureFlags.EnableInactivityCheckJob {
+		mlog.Info("No activity check because EnableInactivityCheckJob feature flag is disabled")
+		return
+	}
+
+	inactivityDurationHourseEnv := os.Getenv("MM_INACTIVITY_DURATION")
+	inactivityDurationHours, parseError := strconv.ParseFloat(inactivityDurationHourseEnv, 64)
+	if parseError != nil {
+		// default to 100 hours
+		inactivityDurationHours = serverInactivityHours
+	}
+
+	systemValue, sysValErr := s.Store.System().GetByName("INACTIVITY")
+	if sysValErr != nil {
+		if _, ok := sysValErr.(*store.ErrNotFound); !ok {
+			mlog.Warn("An error occurred while getting INACTIVITY from system store", mlog.Err(sysValErr))
+		}
+	}
+
+	// If we have a system value, it means this job already ran atleast once.
+	// we then check the last time the job ran plus the last time a post was made to determine if we
+	// can remind the user to use workspace again. If no post was made, we check the last time they logged in (session)
+	// and determine whether to send them a reminder.
+	if systemValue != nil {
+		sysT, _ := strconv.ParseInt(systemValue.Value, 10, 64)
+		tt := time.Unix(sysT/1000, 0)
+		timeLastSentInativityEmail := time.Since(tt).Hours()
+
+		lastPostAt, _ := s.Store.Post().GetLastPostRowCreateAt()
+		if lastPostAt != 0 {
+			posT := time.Unix(lastPostAt/1000, 0)
+			timeForLastPost := time.Since(posT).Hours()
+
+			if timeLastSentInativityEmail > inactivityDurationHours && timeForLastPost > inactivityDurationHours {
+				s.takeInactivityAction()
+			}
+			return
+		}
+
+		lastSessionAt, _ := s.Store.Session().GetLastSessionRowCreateAt()
+		if lastSessionAt != 0 {
+			sesT := time.Unix(lastSessionAt/1000, 0)
+			timeForLastSession := time.Since(sesT).Hours()
+
+			if timeLastSentInativityEmail > inactivityDurationHours && timeForLastSession > inactivityDurationHours {
+				s.takeInactivityAction()
+			}
+			return
+		}
```

Comment:
This seems to be the same functionality as below, any way we can extract this from this `if` and consolidate below?

Reply:
It's not exactly the same, below, there's no `timeLastSentInativityEmail `

Reply:
We can maybe unify this duplicate code by extracting the one different condition into a var (i.e `shouldSendEmail`) which defaults to `true` and can be overriden if it was sent before and meets the condition. What do you think?

```suggestion

	shouldSendEmail := true;

	// Check if the email was sent before and we shouldn't send again 
	if systemValue != nil {
		sysT, _ := strconv.ParseInt(systemValue.Value, 10, 64)
		tt := time.Unix(sysT/1000, 0)
		timeLastSentInativityEmail := time.Since(tt).Hours()

		shouldSendEmail :=  timeLastSentInativityEmail > inactivityDurationHours
	}
	
	// We check if the user has not made any posts and remind them to use the workspace.
	// If no posts have been made. We check the last time
	// they logged in (session) and send a reminder.

	lastPostAt, _ := s.Store.Post().GetLastPostRowCreateAt()
	if lastPostAt != 0 {
		posT := time.Unix(lastPostAt/1000, 0)
		timeForLastPost := time.Since(posT).Hours()
		if shouldSendEmail && timeForLastPost > inactivityDurationHours {
			s.takeInactivityAction()
		}
		return
	}

	lastSessionAt, _ := s.Store.Session().GetLastSessionRowCreateAt()
	if lastSessionAt != 0 {
		sesT := time.Unix(lastSessionAt/1000, 0)
		timeForLastSession := time.Since(sesT).Hours()
		if  shouldSendEmail && timeForLastSession > inactivityDurationHours {
			s.takeInactivityAction()
		}
		return
	}
}
```

Reply:
Thoughts on the above? @AGMETEOR